OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -133,11 +150,15 @@ public sealed class HackyAIInfo : IBotInfo, ITraitInfo
 		public readonly Dictionary<string, float> BuildingFractions = null;
 
 		[Desc("Tells the AI what unit types fall under the same common name. Only supported entry is Mcv.")]
-		public readonly Dictionary<string, HashSet<string>> UnitsCommonNames = null;
+		[FieldLoader.LoadUsing("LoadUnitCategories")]
+		[FieldLoader.Require]
+		public readonly UnitCategories UnitsCommonNames;
 
 		[Desc("Tells the AI what building types fall under the same common name.",
 			"Possible keys are ConstructionYard, Power, Refinery, Silo , Barracks, Production, VehiclesFactory, NavalProduction.")]
-		public readonly Dictionary<string, HashSet<string>> BuildingCommonNames = null;
+		[FieldLoader.LoadUsing("LoadBuildingCategories")]
```

Comment:
We don't have a convention but we could start one here.
Also no real reason other than taking up less space.